hooks/windows/check-arch.js

```diff
@@ -0,0 +1,53 @@
+module.exports = function(ctx) {
+    if (ctx.opts && ctx.opts.platforms && ctx.opts.platforms.indexOf('windows') > -1
+        && ctx.opts.options) {
+        var shell = ctx.requireCordovaModule('shelljs');
+        var path = ctx.requireCordovaModule('path');
+        var nopt = ctx.requireCordovaModule('nopt');
+
+        // parse and validate args
+        var args = nopt({
+            'archs': [String],
+            'appx': String,
+            'phone': Boolean,
+            'win': Boolean,
+            'bundle': Boolean,
+            'packageCertificateKeyFile': String,
+            'packageThumbprint': String,
+            'publisherId': String,
+            'buildConfig': String
+        }, {}, ctx.opts.options.argv, 0);
+
+        // Check if --appx flag is passed so that we have a project build version override:  
+        var isWin10 = args.appx && args.appx.toLowerCase() === 'uap';
+
+        // Else check "windows-target-version" preference:
+        if (!isWin10) {
+            var configXml = shell.ls(path.join(ctx.opts.projectRoot, 'config.xml'))[0];
+
+            var reTargetVersion = /<preference\s+name="windows-target-version"\s+value="(.+)"\s*\/>/i;
+            var targetVersion = shell.grep(reTargetVersion, configXml);
+
+            var result = reTargetVersion.exec(targetVersion);
+            if (result !== null) {
+                var match = result[1];
+                isWin10 = parseInt(match.split('.'), 10) > 8;
+            }
+        }
+
+        // Non-AnyCPU arch is required for Windows 10 (UWP) projects only:
+        if (isWin10) {
+            var rawArchs = ctx.opts.options.archs || args.archs;
+            var archs = rawArchs ? rawArchs.split(' ') : [];
+            // Filter out "anycpu"
+            archs = archs.filter(function (item) {
+                return item !== 'anycpu';
+            });
+
+            if (archs.length === 0) {
```

Comment:
Should we also throw if there are multiple architectures specified _including_ AnyCpu? E.g. `['x86', 'x64', 'anycpu']`